src/Yoti/Entity/ExtraData.php

```diff
@@ -0,0 +1,37 @@
+<?php
+
+namespace Yoti\Entity;
+
+use Yoti\Util\Validation;
+
+class ExtraData
+{
+    /**
+     * @var mixed[]
+     */
+    private $dataEntryList = [];
+
+    /**
+     * @param mixed[] $dataEntryList
+     */
+    public function __construct(array $dataEntryList)
+    {
+        Validation::isArrayOfType($dataEntryList, [AttributeIssuanceDetails::class], 'dataEntryList');
+        $this->dataEntryList = $dataEntryList;
+    }
+
+    /**
+     * @return \Yoti\Entity\AttributeIssuanceDetails|null
+     */
+    public function getAttributeIssuanceDetails()
+    {
+        $attributeIssuanceDetails = array_filter(
+            $this->dataEntryList,
+            function ($dataEntry) {
+                return $dataEntry instanceof AttributeIssuanceDetails;
+            }
+        );
+
+        return reset($attributeIssuanceDetails) ?: null;
+    }
```

Comment:
When I spoke to Mike about this in Java, I think it was preferred to do the finding up front and only once.  This way we don't have to search for the `AttributeIssuanceDetails` multiple times if the user calls this method more than once?  What do you think?

Reply:
Fair point, but can you envisage clients needing to call it more than once? It would probably be a code smell if they were calling the same method twice wouldn't it? (when nothing else has changed)

Reply:
👍 Agree - optimised in bc018e8